lib/spotify/responder.ex

```diff
@@ -12,8 +12,29 @@ defmodule Responder do
       def handle_response({ :ok, %HTTPoison.Response{ status_code: code, body: "" }})
         when code in 200..299, do: :ok
 
-      def handle_response({ message, %HTTPoison.Response{ status_code: code, body: body }})
-        when code in 400..499, do: { message, Poison.decode!(body)}
+      # special handling for 'too many requests' status
+      # in order to know when to retry
+      def handle_response(
+        { message, %HTTPoison.Response{ status_code: 429, body: body, headers: headers }}
+      ) do
+        {retry_after, ""} = (
+          headers
+          |> Enum.find(&(Kernel.elem(&1, 0) == "Retry-After"))
```

Comment:
Here we're assuming a 429 response code will always have a JSON Object in the response body *and* a `Retry-After` in the headers.  This is consistent with my experience with the behavior of the API so far.